src/Language/PureScript/Make.hs

```diff
@@ -221,15 +221,16 @@ make MakeActions{..} ms = do
                               _ -> True
 
         let rebuild = do
-              (exts, warnings) <- listen $ do
+              (exts, warnings) <- listen $ censor suppressHoleWarnings $ do
```

Comment:
I would just add a line [here](https://github.com/purescript/purescript/blob/dc58903b8a19af7cb44b63ead3295e6133b740f5/src/Language/PureScript/TypeChecker/Types.hs#L120) and then throw during codegen if you see the `Hole` constructor.